staging/src/k8s.io/apiserver/pkg/server/filters/with_retry_after.go

```diff
@@ -62,13 +62,25 @@ type ShouldRespondWithRetryAfterFunc func() (*RetryAfterParams, bool)
 //   - 'Connection: close': tear down the TCP connection
 //
 // TODO: is there a way to merge WithWaitGroup and this filter?
-func WithRetryAfter(handler http.Handler, when ShouldRespondWithRetryAfterFunc) http.Handler {
+func WithRetryAfter(handler http.Handler, shutdownDelayDurationElapsedCh <-chan struct{}) http.Handler {
+	shutdownRetryAfterParams := &retryAfterParams{
+		TearDownConnection: true,
```

Comment:
this might be unnecessary on an SNO cluster

Reply:
but should not break anything - we just might create a few more new con

Reply:
same, this is part of the pick from upstream

Reply:
sure, just thinking out loud, now i recall it will be retried by the client in an exponential way